ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CompletionStorageBackendTemplate.scala

```diff
@@ -150,4 +154,21 @@ trait CompletionStorageBackendTemplate extends CompletionStorageBackend {
     rejectionStatusDetails
       .map(stream => StatusDetails.parseFrom(stream).details)
       .getOrElse(Seq.empty)
+
+  def pruneCompletions(
+      pruneUpToInclusive: Offset
+  )(connection: Connection, loggingContext: LoggingContext): Unit = {
+    pruneWithLogging(queryDescription = "Command completions pruning") {
+      import com.daml.platform.store.Conversions.OffsetToStatement
+      SQL"delete from participant_command_completions where completion_offset <= $pruneUpToInclusive"
+    }(connection, loggingContext)
+  }
+
+  private def pruneWithLogging(queryDescription: String)(query: SimpleSql[Row])(
```

Comment:
This method is now duplicated between CompletionStorageBackendTemplate and EventStorageBackendTemplate, both of these traits implement pruning methods.